**List of Imports**

In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

import pandas as pd
import numpy as np
import matplotlib as mpl
from matplotlib import pyplot as plt
import seaborn as sns
import warnings
from collections import Counter
import datetime
import wordcloud
import json
%matplotlib inline
%config InlineBackend.figure_format = 'retina'

#Making the output look cleaner
warnings.filterwarnings('ignore')

**Reading the Dataset**

In [2]:
#Read the Youtube Dataset for USA
df = pd.read_csv("/home/sleepynav/Documents/Social Web/Youtube-Trending/Data/USvideos.csv", index_col='video_id')

with open("/home/sleepynav/Documents/Social Web/Youtube-Trending/Data/US_category_id.json") as f:
    categories = json.load(f)["items"]
cat_dict = {}

for cat in categories:
    cat_dict[int(cat["id"])] = cat["snippet"]["title"]
df['category_name'] = df['category_id'].map(cat_dict)

**Exploring the CSV File**

In [3]:
df.head(1)

trending_date                               title channel_title  \
video_id                                                                      
2kyS6SvSYSE      17.14.11  WE WANT TO TALK ABOUT OUR MARRIAGE  CaseyNeistat   

             category_id              publish_time             tags   views  \
video_id                                                                      
2kyS6SvSYSE           22  2017-11-13T17:13:01.000Z  SHANtell martin  748374   

             likes  dislikes  comment_count  \
video_id                                      
2kyS6SvSYSE  57527      2966          15954   

                                             thumbnail_link  \
video_id                                                      
2kyS6SvSYSE  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   

             comments_disabled  ratings_disabled  video_error_or_removed  \
video_id                                                                   
2kyS6SvSYSE              False             False                   False   

                                                   description   category_name  
video_id                                                                        
2kyS6SvSYSE  SHANTELL'S CHANNEL - https://www.youtube.com/s...  People & Blogs

**Exploring the different Categories set**

In [4]:
cat_dict

{1: 'Film & Animation',
 2: 'Autos & Vehicles',
 10: 'Music',
 15: 'Pets & Animals',
 17: 'Sports',
 18: 'Short Movies',
 19: 'Travel & Events',
 20: 'Gaming',
 21: 'Videoblogging',
 22: 'People & Blogs',
 23: 'Comedy',
 24: 'Entertainment',
 25: 'News & Politics',
 26: 'Howto & Style',
 27: 'Education',
 28: 'Science & Technology',
 29: 'Nonprofits & Activism',
 30: 'Movies',
 31: 'Anime/Animation',
 32: 'Action/Adventure',
 33: 'Classics',
 34: 'Comedy',
 35: 'Documentary',
 36: 'Drama',
 37: 'Family',
 38: 'Foreign',
 39: 'Horror',
 40: 'Sci-Fi/Fantasy',
 41: 'Thriller',
 42: 'Shorts',
 43: 'Shows',
 44: 'Trailers'}

** Techinical information about the dataset **

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 40949 entries, 2kyS6SvSYSE to ooyjaVdt-jA
Data columns (total 16 columns):
trending_date             40949 non-null object
title                     40949 non-null object
channel_title             40949 non-null object
category_id               40949 non-null int64
publish_time              40949 non-null object
tags                      40949 non-null object
views                     40949 non-null int64
likes                     40949 non-null int64
dislikes                  40949 non-null int64
comment_count             40949 non-null int64
thumbnail_link            40949 non-null object
comments_disabled         40949 non-null bool
ratings_disabled          40949 non-null bool
video_error_or_removed    40949 non-null bool
description               40379 non-null object
category_name             40949 non-null object
dtypes: bool(3), int64(5), object(8)
memory usage: 4.5+ MB


In [8]:
df['trending_date'] = pd.to_datetime(df['trending_date'], format='%y.%d.%m')
df['trending_date'].head()

video_id
2kyS6SvSYSE   2017-11-14
1ZAPwfrtAFY   2017-11-14
5qpjK5DgCt4   2017-11-14
puqaWrEC7tY   2017-11-14
d380meD0W0M   2017-11-14
Name: trending_date, dtype: datetime64[ns]

In [9]:
df['publish_time'] = pd.to_datetime(df['publish_time'], format='%Y-%m-%dT%H:%M:%S.%fZ')
df['publish_time'].head()

video_id
2kyS6SvSYSE   2017-11-13 17:13:01
1ZAPwfrtAFY   2017-11-13 07:30:00
5qpjK5DgCt4   2017-11-12 19:05:24
puqaWrEC7tY   2017-11-13 11:00:04
d380meD0W0M   2017-11-12 18:01:41
Name: publish_time, dtype: datetime64[ns]

** Seperating the Publish columns of Date and Time for Better Clarity **

In [10]:
df.insert(4, 'publish_date', df['publish_time'].dt.date)
df['publish_time'] = df['publish_time'].dt.time
df[['publish_date', 'publish_time']].head()

publish_date publish_time
video_id                             
2kyS6SvSYSE   2017-11-13     17:13:01
1ZAPwfrtAFY   2017-11-13     07:30:00
5qpjK5DgCt4   2017-11-12     19:05:24
puqaWrEC7tY   2017-11-13     11:00:04
d380meD0W0M   2017-11-12     18:01:41

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 40949 entries, 2kyS6SvSYSE to ooyjaVdt-jA
Data columns (total 17 columns):
trending_date             40949 non-null datetime64[ns]
title                     40949 non-null object
channel_title             40949 non-null object
category_id               40949 non-null int64
publish_date              40949 non-null object
publish_time              40949 non-null object
tags                      40949 non-null object
views                     40949 non-null int64
likes                     40949 non-null int64
dislikes                  40949 non-null int64
comment_count             40949 non-null int64
thumbnail_link            40949 non-null object
comments_disabled         40949 non-null bool
ratings_disabled          40949 non-null bool
video_error_or_removed    40949 non-null bool
description               40379 non-null object
category_name             40949 non-null object
dtypes: bool(3), datetime64[ns](1), int64(5), object(8)
memory usag

**It is clear with the information on the data set that all columns but one contain 40949 non null objects.**

**Organizing and Cleaning the data for Description column that has null objects**

In [12]:
df[df["description"].apply(lambda x: pd.isnull(x))].head(3)

trending_date                                              title  \
video_id                                                                       
NZFhMSgbKKM    2017-11-14  Dennis Smith Jr. and LeBron James go back and ...   
sbcbvuitiTc    2017-11-14  Stephon Marbury and Jimmer Fredette fight in C...   
4d07RXYLsJE    2017-11-14                Sphaera - demonstrating interaction   

                        channel_title  category_id publish_date publish_time  \
video_id                                                                       
NZFhMSgbKKM              Ben Rohrbach           17   2017-11-13     15:11:00   
sbcbvuitiTc  NBA Highlights · YouTube           17   2017-11-10     18:23:05   
4d07RXYLsJE              Jenny Hanell           28   2017-11-04     20:48:16   

                                  tags   views  likes  dislikes  \
video_id                                                          
NZFhMSgbKKM                     [none]     945      7         5   
sbcbvuitiTc  NBA|"Basketball"|"Sports"  956169   2017      2425   
4d07RXYLsJE                     [none]    1827      3         0   

             comment_count                                  thumbnail_link  \
video_id                                                                     
NZFhMSgbKKM              8  https://i.ytimg.com/vi/NZFhMSgbKKM/default.jpg   
sbcbvuitiTc           1447  https://i.ytimg.com/vi/sbcbvuitiTc/default.jpg   
4d07RXYLsJE              2  https://i.ytimg.com/vi/4d07RXYLsJE/default.jpg   

             comments_disabled  ratings_disabled  video_error_or_removed  \
video_id                                                                   
NZFhMSgbKKM              False             False                   False   
sbcbvuitiTc              False             False                   False   
4d07RXYLsJE              False             False                   False   

            description         category_name  
video_id                                       
NZFhMSgbKKM         NaN                Sports  
sbcbvuitiTc         NaN                Sports  
4d07RXYLsJE         NaN  Science & Technology

**Fill null values in description with an empty string**

In [13]:
df["description"] = df["description"].fillna(value="")
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 40949 entries, 2kyS6SvSYSE to ooyjaVdt-jA
Data columns (total 17 columns):
trending_date             40949 non-null datetime64[ns]
title                     40949 non-null object
channel_title             40949 non-null object
category_id               40949 non-null int64
publish_date              40949 non-null object
publish_time              40949 non-null object
tags                      40949 non-null object
views                     40949 non-null int64
likes                     40949 non-null int64
dislikes                  40949 non-null int64
comment_count             40949 non-null int64
thumbnail_link            40949 non-null object
comments_disabled         40949 non-null bool
ratings_disabled          40949 non-null bool
video_error_or_removed    40949 non-null bool
description               40949 non-null object
category_name             40949 non-null object
dtypes: bool(3), datetime64[ns](1), int64(5), object(8)
memory usag

**As we can see now all columns have equal number of non null rows i.e 40949**

In [14]:
df.describe()

category_id         views         likes      dislikes  comment_count
count  40949.000000  4.094900e+04  4.094900e+04  4.094900e+04   4.094900e+04
mean      19.972429  2.360785e+06  7.426670e+04  3.711401e+03   8.446804e+03
std        7.568327  7.394114e+06  2.288853e+05  2.902971e+04   3.743049e+04
min        1.000000  5.490000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       17.000000  2.423290e+05  5.424000e+03  2.020000e+02   6.140000e+02
50%       24.000000  6.818610e+05  1.809100e+04  6.310000e+02   1.856000e+03
75%       25.000000  1.823157e+06  5.541700e+04  1.938000e+03   5.755000e+03
max       43.000000  2.252119e+08  5.613827e+06  1.674420e+06   1.361580e+06